### DISCUSSION TITLE
Reuse Postgres custom enum type?

### DISCUSSION THREAD

**Discussion by g-bar** [sqlalchemy/alembic#1070] [Usage Questions]
So I have 2 models that use the same custom Enum type like so:

```python
from sqlalchemy.dialects.postgresql import ENUM

class ModelA(Base):
    __tablename__ = "model_a"
    type = Column(ENUM("one", "two", name="structure_type"), nullable=False)

class ModelB(Base):
    __tablename__ = "model_b"
    type = Column(ENUM("one", "two", name="structure_type"), nullable=False)

```

After autogenerating the migration script and trying to upgrade the schema I get an error:

```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateObject) type "structure_type" already exists

[SQL: CREATE TYPE structure_type AS ENUM ('membrane', 'compartment')]
```

It looks like Alembic generated 2 separate `CREATE TYPE` statements with the same name. The easiest work around is to name the second one differently but that would result in 2 equal types in the database. Is there a way to tell alembic to not create a new type but use an existing one

Thanks for the input.

**Comment by zzzeek** [comment]
pass [create_type=False](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html?highlight=postgresql+enum#sqlalchemy.dialects.postgresql.ENUM.params.create_type) to the ENUM constructor within your alembic migration scripts, then use the [create()](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html?highlight=enum+create#sqlalchemy.dialects.postgresql.ENUM.create) method in your alembic migration as needed i.e. `myenum.create(bind=op.get_bind())`.
